wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java

```diff
@@ -10,158 +10,204 @@
-     *         when given user cannot be updated
+     *         when user doesn't exist
      * @throws ServerException
      *         when any other error occurs
-     *
      */
-    public void update(User user) throws NotFoundException, ServerException, ConflictException {
-        userDao.update(user);
+    public User getByAlias(String alias) throws NotFoundException, ServerException {
+        requireNonNull(alias, "Required non-null alias");
+        return userDao.getByAlias(alias);
     }
 
-
     /**
-     * Gets user from persistent layer by any of his aliases
+     * Finds user by given {@code name}.
      *
-     * @param alias
-     *         user name or alias
-     * @return user POJO
+     * @param name
+     *         user name
+     * @return user instance
+     * @throws NullPointerException
+     *         when {@code name} is null
      * @throws NotFoundException
      *         when user doesn't exist
      * @throws ServerException
      *         when any other error occurs
      */
-    public User getByAlias(String alias) throws NotFoundException, ServerException {
-        return userDao.getByAlias(alias);
+    public User getByName(String name) throws NotFoundException, ServerException {
+        requireNonNull(name, "Required non-null name");
+        return userDao.getByName(name);
     }
 
-
     /**
-     * Gets user from persistent layer by his username
+     * Finds user by given {@code email}.
      *
-     * @param userName
-     *         user name
-     * @return user POJO
+     * @param email
+     *         user email
+     * @return user instance
+     * @throws NullPointerException
+     *         when {@code email} is null
      * @throws NotFoundException
      *         when user doesn't exist
      * @throws ServerException
      *         when any other error occurs
      */
-    public User getByName(String userName) throws NotFoundException, ServerException {
-        return userDao.getByName(userName);
+    public User getByEmail(String email) throws NotFoundException, ServerException {
+        requireNonNull(email, "Required non-null email");
+        return userDao.getByEmail(email);
     }
 
-
     /**
-     * Removes user from persistent layer by his identifier.
+     * Removes user and his dependencies by given {@code id}.
      *
      * @param id
      *         user identifier
+     * @throws NullPointerException
+     *         when {@code id} is null
      * @throws ConflictException
      *         when given user cannot be deleted
      * @throws ServerException
      *         when any other error occurs
      */
-    public void remove(String id) throws NotFoundException, ServerException, ConflictException {
+    public void remove(String id) throws ServerException, ConflictException {
+        requireNonNull(id, "Required non-null id");
+        profileManager.remove(id);
+        preferencesManager.remove(id);
         userDao.remove(id);
     }
 }
```

Comment:
Not sure I support data integrity operations on this level (I take Manager as business level entity) especially calling one Manager from another.
Could you at least replace profileManager/preferencesManager.remove() with profileDao/preferencesDao.remove() ?